oap-ape/ape-java/ape-spark/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetRecordReaderWrapperFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet;
+
+public class ParquetRecordReaderWrapperFactory {
+  public static ParquetRecordReaderWrapper getParquetRecordReaderWrapper(
+          String parquetRecordReaderWrapperType, int capacity) {
+    if (parquetRecordReaderWrapperType == "local") {
+      return new ParquetNativeRecordReaderWrapper(capacity);
+    } else if (parquetRecordReaderWrapperType == "remote") {
+      return new ParquetRemoteRecordReaderWrapper(capacity);
+    }
+    return null;
```

Comment:
throw an exception instead?  if a user set a wrong parameter, an meaningful exception will be much easier to debug compare to a NullPointerExcption